server/src/entities/asset.entity.ts

```diff
@@ -170,6 +168,13 @@ export class AssetEntity {
   @OneToOne(() => AssetJobStatusEntity, (jobStatus) => jobStatus.asset, { nullable: true })
   jobStatus?: AssetJobStatusEntity;
 
+  @Column({
+    type: 'enum',
+    enum: AssetTrashReason,
+    nullable: true,
+  })
+  trashReason!: AssetTrashReason | null;
```

Comment:
I think it makes sense for it to be nullable, personally. This is something that's only really known when it's being trashed, so it should be set at that time.

Reply:
Right, so it should be set to deleted. Everything that exists right now is deleted. What's the reason for not setting it?

Reply:
You bring up a good point @zackpollard. Any existing assets that are in the trash must be migrated to have trashReason set to deleted.

I also agree with @mertalev that trashReason should be nullable since the field will be null until it's trashed.

Therefore, I need to write a migration to set the field for all deleted assets.

Reply:
Are you talking about the migration? The value it should be set to depends on whether `deletedAt` or `isOffline` is true, so there should be a query that sets it with `case when "deletedAt" is not null then 'deleted' when "isOffline" then 'offline' end`. In the case where `isOffline` is true and `deletedAt` is null, it should also set `deletedAt`.

Reply:
Honestly completely overlooked the case of it not being trashed 🤣 But yes we can do that for the existing trashed assets was more my point :P

Reply:
The question seems to be "do we want `null` to imply `deleted`". I am fine making it non-nullable, always requiring a value, and setting it to `deleted` when a request comes in through the api.

Reply:
With kysely, null fields can be omitted from DB responses entirely, so it's generally better to set something null than to have a default that has the same meaning.

Reply:
Very free queries return deleted data in the first place, so I don't think that's a very strong argument.

Reply:
But if `trashReason` were always set to a value, it'd always be part of the response even if the asset isn't trashed. I guess you could omit it from the select instead, but that's not as clean.

Reply:
Oh right. I think I meant to recommend it always have a value _when an asset has been deleted_. So null when not deleted, deleted when trashed and offline when it goes offline. I guess that means nullable, but not "null implies deleted".

Reply:
Ah, that makes sense. I also agree with that.

Reply:
Maybe there should be a check constraint on this like `("deletedAt" is null) = ("trashReason" is null)`, so we have an invariant that either both are null or neither are.

Reply:
Just to confirm: I currently implemented this by having trashReason be null for all non-trashed item. When an item is trashed by the api, it's set to deleted, and when the file is offline it's set to offline. This is the way we want, right?

Reply:
Yes that sounds right to me 🙂